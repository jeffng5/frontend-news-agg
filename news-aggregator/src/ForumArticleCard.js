import React from 'react'
import './css/forum.css'
import Comments from './Comments'
import Comment from './Comment'

// component to add structure into each article 
const ForumArticleCard = ({title, description, url, author, likes, id}) =>{
    return (
        <>
        <p className='forum-article-card'>
        <a href= {url} className= 'url' target='_blank'><p className='title-forum-article'>{title}</p><p className= 'description-forum-article'>{description}</p></a><p className='author-forum-article'>-Author: {author}</p>
{likes}
        
        <p className='commentary'><Comments id ={id} /></p> 
        <p className= 'commentary'><Comment id={id}/></p>
        </p>
        </>
    
    )
}


export default ForumArticleCard