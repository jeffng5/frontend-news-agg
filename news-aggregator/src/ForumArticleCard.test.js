import React from 'react';
import { render } from '@testing-library/react';
import ForumArticleCard from './ForumArticleCard'

it('renders without crashing', function() {
  render(<ForumArticleCard />);
})

//snapshot test
it("matches snapshot", function() {
    const {asFragment} = render(<ForumArticleCard />);
    expect(asFragment()).toMatchSnapshot();
  });

test('testing component', ()=> {
    const {getByText} = render( <ForumArticleCard />);
    getByText('Author:')
})