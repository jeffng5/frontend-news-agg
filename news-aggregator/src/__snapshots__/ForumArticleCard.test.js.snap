// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <p
    class="forum-article-card"
  >
    <a
      class="url"
      target="_blank"
    >
      <h7
        class="title-forum-article"
      />
      <h7
        class="description-forum-article"
      />
    </a>
    <h7
      class="author-forum-article"
    >
      -Author: 
    </h7>
  </p>
  <p
    class="commentary"
  />
  <p
    class="commentary"
  />
  <form>
    <textarea
      cols="50"
      name="message"
      placeholder="Type comment here..."
      rows="5"
    />
  </form>
  <button
    class="add-comment"
  >
    Add Comment
  </button>
  <h4 />
  <p />
  <p />
</DocumentFragment>
`;
